### PR TITLE
Add custom exception handling layer

### PR DESCRIPTION
**Pull request by mofr**
Custom handling layer is added. Main goals:
* Allow users to handle custom exceptions without using DRF or Django
* Generate swagger for exceptions

### CODE REVIEW

File: winter/django.py [hunk 1]
```diff
@@ -71,28 +75,45 @@ class WinterView(rest_framework.views.APIView):
...
+
     return dispatch
 
 
+def _call_controller_method(controller, controller_method: ControllerMethod, path_variables: Dict, request: Request):
+    arguments = arguments_resolver.resolve_arguments(controller_method, request, path_variables)
+    try:
+        result = controller_method.func(controller, **arguments)
+        return _convert_result_to_http_response(request, result, controller_method.func)
+    except Exception as e:
```
Comment (berenda-andrey) [inline winter/django.py]: You can use 
`
except tuple(get_throws(func)) as e

`
And delete is_expected_to_throw

File: winter/exceptions.py [hunk 2]
```diff
@@ -18,3 +28,60 @@ def handle_winter_exception(exception: WinterException) -> HTTPResponse:
...
+def mark_func_throws(func, exception_cls: Type[Exception]):
+    _throws[func].add(exception_cls)
+
+
+def get_throws(func) -> Optional[Set[Type[Exception]]]:
+    return _throws.get(func)
+
+
+def is_expected_to_throw(func, e: Exception) -> bool:
+    expected_exception_cls_set = get_throws(func) or set()
```
Comment (berenda-andrey) [inline winter/exceptions.py]: delete or set

File: tests/controllers/controller_with_exceptions.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import dataclasses
+from rest_framework.request import Request
+
+import winter
+
+
+class CustomException(Exception):
+    def __init__(self, message: str):
+        self.message = message
```
Comment (pristupa) [inline tests/controllers/controller_with_exceptions.py]: super().__init__(message)

File: tests/controllers/controller_with_exceptions.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+
+@winter.controller
+@winter.route('controller_with_exceptions/')
+class ControllerWithExceptions:
+
+    @winter.route_get('declared_but_not_thrown/')
+    @winter.throws(CustomException)
+    def declared_but_not_thrown(self) -> str:
+        return f'Hello, sir!'
```
Comment (pristupa) [inline tests/controllers/controller_with_exceptions.py]: no "f"

File: tests/schema/test_response_schema.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
+from typing import Dict
+
+import pytest
+from drf_yasg import openapi
+
+from winter.controller import ControllerMethod
+
```
Comment (pristupa) [inline tests/schema/test_response_schema.py]: remove blank line

File: winter/exceptions.py [hunk 6]
```diff
@@ -18,3 +32,59 @@ def handle_winter_exception(exception: WinterException) -> HTTPResponse:
...
+
+def mark_func_throws(func, exception_cls: Type[Exception]):
+    _throws[func].add(exception_cls)
+
+
+def get_throws(func) -> Optional[Set[Type[Exception]]]:
+    return _throws[func]
+
+
+class ExceptionsHandler(ExceptionHandler):
```
Comment (pristupa) [inline winter/exceptions.py]: ExceptionHandler

**Pull request merged**